### PR TITLE
[FEAT] Added unit tests setup

### LINKED ISSUE #44: [FEAT] Set up unit tests structure

### PR DESCRIPTION
**Pull request by cecilia-yin**
Closes #44

### CODE REVIEW

**Review by David-Xia0 — APPROVED** [review]
Nice, job finally got it running 😄

**Comment by blakehattingh** [comment]
hey Cecilia the code looks great. Good job fixing the issue with the mock index. I think you have forgotten to add labels to the pull request as specified in the PR template. Once that is fixed i will approve

**Comment by blakehattingh** [comment]
approved, nice work

File: backend/src/routes/__tests__/sessions.test.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  mockSession = {
+    'preferences': {
+      'location': 'Auckland',
+      'distance': 10,
+      'cuisines': [
+        'Thai',
+        'Japanese',
+        'Chinese',
+      ],
+      'price': 20,
```
Comment (blakehattingh) [inline backend/src/routes/__tests__/sessions.test.js]: I'm pretty sure this price is supposed to be between 1 and 4. Is this testing for a bug intentionally? if not please change to the accepted range

**Review by lilichenyuxin — APPROVED** [review]

**Comment by lilichenyuxin** [comment]
LGTM, nice work cici

**Review by Jaysce — APPROVED** [review]
LGTM

**Review by blakehattingh — DISMISSED** [review]
legend, nice work

**Review by Jaysce — APPROVED** [review]
This is some clean code. I love clean code. 🧼

**Comment by Alxjlv** [comment]
There's some stuff that's gone wrong with the testing. For one - the mocks in dist need to be ignored in jest:
image

Also, it appears that the tests are leaving some sockets disconnected and hanging open, which leaves the tests running continuously:
image

**Comment by Alxjlv** [comment]
Maybe adding the `--detectOpenHandles` to jest in the package.json backend test script could help to troubleshoot why the test is hanging

**Review by Alxjlv — CHANGES_REQUESTED** [review]
The test suite needs to be fixed so that it doesn't hang for 20+ minutes

File: backend/package.json [hunk 2]
```diff
@@ -4,16 +4,23 @@
...
+  "jest": {
+    "testEnvironment": "node",
+    "testPathIgnorePatterns": [
+      "<rootDir>/dist/"
+    ]
+  },
   "keywords": [],
   "author": "",
   "dependencies": {
+    "axios": "^0.21.1",
```
Comment (Jaysce) [inline backend/package.json:23]: Maybe change this so that its a dev dependency instead.

**Review by Jaysce — APPROVED** [review]
Look good! Nice 👍

**Review by lilichenyuxin — APPROVED** [review]

**Comment by lilichenyuxin** [comment]
LGTM

**Review by Jaysce — APPROVED** [review]
Changes have been done.

**Review by BlinkCreator — APPROVED** [review]
looks good

**Review by Alxjlv — APPROVED** [review]
👍

**Pull request merged**